helm-chart/renku-notebooks/values.yaml

```diff
@@ -3,13 +3,19 @@
 # Declare variables to be passed into your templates.
 
 global:
-  useHTTPS: false
+  ## Specify origin and service prefix (for example /gitlab)
+  ## for the GitLab service.
   gitlab:
-    urlPrefix: /
+    origin: # https://gitlab.com
```

Comment:
The goal of this exercise is to put things like the URL for gitlab which many components need into the global chart section and to put them in a form which all renku services can understand. Traefik for example insists in having this distinction (it just strips the path from the URL of a backend server).

Reply:
To follow up on this topic: Go templates have limited regex support (http://masterminds.github.io/sprig/strings.html) which makes it a cumbersome task to split a URL into its part. Therefore, I thought that providing the origin (<scheme> "://" <hostname> [ ":" <port> ]) and the path prefix for each component would be easier. However, I just checked again, and I think through a combination of `regexFind` and `trimPrefix` it would be possible to parse a full URL inside templates. This will make the `values.yaml` files simpler and the templates a bit more complicated, which is also a viable option for me.